src/trailer.c

```diff
@@ -0,0 +1,403 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#include "common.h"
+#include "git2/message.h"
+
+#include <stddef.h>
+#include <string.h>
+#include <ctype.h>
+
+static const char comment_line_char = '#';
+
+static const char *const git_generated_prefixes[] = {
+	"Signed-off-by: ",
+	"(cherry picked from commit ",
```

Comment:
Is this really a trailer line? I thought the definition of a trailer line is that it is of the form "key: value"

Reply:
I think this is one of those git-core-specific intricacies I was talking about. I think technically a trailer can be in just about any configurable format (there are git-config settings we'll need to implement support for here, eventually). The HTTP-style `Key: value` stuff is just the default.

It looks like this stuff was added in git-core in https://github.com/git/git/commit/146245063e286d5a38b146bd5a38da958bd3957b - forgot there was also a heuristic used to determine what a trailer block is...